backend/onedrive/onedrive.go

```diff
@@ -1715,12 +1784,16 @@ func (o *Object) uploadSinglepart(ctx context.Context, in io.Reader, size int64,
 	var resp *http.Response
 	var opts rest.Opts
 	leaf, directoryID, _ := o.fs.dirCache.FindPath(ctx, o.remote, false)
-	trueDirID, drive, rootURL := parseNormalizedID(directoryID)
+	trueDirID, drive, rootURL := parseNormalizedID(directoryID, o.fs.opt.Region)
 	if drive != "" {
+		path := "/" + drive + "/items/" + trueDirID + ":/" + rest.URLPathEscape(o.fs.opt.Enc.FromStandardName(leaf)) + ":/content"
+		if o.fs.opt.Region == "cn" {
+			path = "/" + drive + "/items/" + trueDirID + "/children/" + rest.URLPathEscape(o.fs.opt.Enc.FromStandardName(leaf)) + "/content"
```

Comment:
Does this mean the request will fail if the file is named `content`?

Reply:
I tested this case and the request succeeded.
```
[toki@xytoki-vm rclone]$ ./rclone copy local:/home/toki/content od:/ -vvv
2020/02/18 22:47:07 DEBUG : rclone: Version "v1.51.0-DEV" starting with parameters ["./rclone" "copy" "local:/root/content" "od:/" "-vvv"]
2020/02/18 22:47:07 DEBUG : Using config file from "/home/toki/.config/rclone/rclone.conf"
2020/02/18 22:47:08 DEBUG : content: Need to transfer - File not found at Destination
2020/02/18 22:47:08 DEBUG : content: Starting singlepart upload
2020/02/18 22:47:09 DEBUG : content: QuickXorHash = 0000000000000000000000000000000000000000 OK
2020/02/18 22:47:09 INFO  : content: Copied (new)
2020/02/18 22:47:09 INFO  : 
Transferred:             0 / 0 Bytes, -, 0 Bytes/s, ETA -
Transferred:            1 / 1, 100%
Elapsed time:         0.8s

[toki@xytoki-vm rclone]$ ./rclone ls od:/ | grep content
        0 content
```